src/main/java/com/redhat/syseng/soleng/rhpam/processmigration/service/impl/PlanServiceImpl.java

```diff
@@ -3,36 +3,60 @@
 import java.util.List;
 
 import javax.enterprise.context.ApplicationScoped;
-import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.Query;
+import javax.transaction.Transactional;
 
-import com.redhat.syseng.soleng.rhpam.processmigration.model.Plan;
-import com.redhat.syseng.soleng.rhpam.processmigration.service.PersistenceService;
+import com.redhat.syseng.soleng.rhpam.processmigration.jpa.model.Plan;
 import com.redhat.syseng.soleng.rhpam.processmigration.service.PlanService;
 
 @ApplicationScoped
 public class PlanServiceImpl implements PlanService {
 
-    @Inject
-    private PersistenceService<Plan> planPersistenceService;
+    @PersistenceContext(unitName = "migration-unit")
+    EntityManager em;
+
+    //Used by Junit test where em can't be injected
+    public void setEntityManager(EntityManager em) {
+        this.em = em;
+    }
 
     @Override
     public Plan get(Long id) {
-	return planPersistenceService.get(id);
+        Plan result = null;
+        Query query = em.createNamedQuery("Plan.findById", Plan.class);
+        query.setParameter("id", id);
+        List<Plan> plans = query.getResultList();
+        for (Plan p : plans) {
+            result = p;
+        }
+
+        return result;
+        //return planPersistenceService.get(id);
     }
 
     @Override
     public List<Plan> findAll() {
-	return planPersistenceService.findAll();
+        List<Plan> plans = em.createNamedQuery("Plan.findAll", Plan.class).getResultList();
+        return plans;
```

Comment:
Remove commented line and return directly the result of `getResultList` without declaring another variable